internal/controller/services/gateway/gateway_controller_actions.go

```diff
@@ -58,52 +59,246 @@ func createGatewayInfrastructure(ctx context.Context, rr *odhtypes.Reconciliatio
+		"domain", hostname,
 		"certificateType", getCertificateType(gatewayConfig))
 
 	return nil
 }
 
-// createDestinationRule creates a DestinationRule for TLS configuration using embedded YAML template.
-// This function uses embedded resources for efficient template management.
-func createDestinationRule(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
-	// Input validation
-	if rr == nil {
-		return errors.New("reconciliation request cannot be nil")
+// Check authentication mode and deploy auth proxy (secret + service + deployment) + OAuth client (if integrated mode) + HTTPRoute + DestinationRule.
+func createKubeAuthProxyInfrastructure(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
+	l := logf.FromContext(ctx).WithName("createAuthProxy")
+
+	gatewayConfig, err := validateGatewayConfig(rr)
+	if err != nil {
+		return err
 	}
 
-	l := logf.FromContext(ctx).WithName("createDestinationRule")
-	l.V(1).Info("Creating DestinationRule for TLS configuration")
+	l.V(1).Info("creating auth proxy for gateway", "gateway", gatewayConfig.Name)
 
-	yamlContent, err := gatewayResources.ReadFile(destinationRuleTemplate)
+	authMode, err := cluster.GetClusterAuthenticationMode(ctx, rr.Client)
 	if err != nil {
-		return fmt.Errorf("failed to read DestinationRule template: %w", err)
+		return fmt.Errorf("failed to detect cluster authentication mode: %w", err)
+	}
+	l.V(1).Info("detected cluster authentication mode", "mode", authMode)
+
+	// Reset the condition to false
+	rr.Conditions.MarkFalse(ReadyConditionType)
+
+	kubeAuthProxyDeploymentTemplates := odhtypes.TemplateInfo{
+		FS:   gatewayResources,
+		Path: kubeAuthProxyDeploymentOauthTemplate,
 	}
 
-	decoder := serializer.NewCodecFactory(rr.Client.Scheme()).UniversalDeserializer()
-	unstructuredObjects, err := resources.Decode(decoder, yamlContent)
+	var oidcConfig *serviceApi.OIDCConfig
+	switch authMode {
+	case cluster.AuthModeOIDC:
+		if gatewayConfig.Spec.OIDC == nil {
+			rr.Conditions.MarkFalse(
+				ReadyConditionType,
+				conditions.WithReason(status.NotReadyReason),
+				conditions.WithMessage(status.AuthProxyOIDCModeWithoutConfigMessage),
+			)
+			return nil // TODO: is this logic correct? no oidc but to use oidc and not error out.
+		}
+		oidcConfig = gatewayConfig.Spec.OIDC
+		l.V(1).Info("configuring "+KubeAuthProxyName+" for external OIDC",
+			"issuerURL", oidcConfig.IssuerURL,
+			"clientID", oidcConfig.ClientID,
+			"secretRef", oidcConfig.ClientSecretRef.Name)
+		kubeAuthProxyDeploymentTemplates = odhtypes.TemplateInfo{
+			FS:   gatewayResources,
+			Path: kubeAuthProxyDeploymentOidcTemplate,
+		}
+	case cluster.AuthModeIntegratedOAuth: // default mode.
+		l.V(1).Info("configuring " + KubeAuthProxyName + " for OpenShift OAuth")
+
+	case cluster.AuthModeNone:
+		rr.Conditions.MarkTrue(
+			ReadyConditionType,
+			conditions.WithReason(status.ReadyReason), // TODO: is this logic correct? user do not want it, we should mark it as ready.
```

Comment:
This is semantically correct. When cluster uses external auth, the desired state is "do not deploy auth proxy", so marking `Ready=True` is appropriate.

```suggestion
			// Ready=True is correct: desired state is "no auth proxy" when using external auth.
			// The status message clarifies: "Cluster uses external authentication, no gateway auth proxy deployed"
  			conditions.WithReason(status.ReadyReason), 
```